addons/mail/static/src/messaging/component/chat_window_manager/chat_window_manager_tests.js

```diff
@@ -845,6 +845,108 @@ QUnit.test('open 2 different chat windows: enough screen width', async function
     );
 });
 
+QUnit.test('open 2 chat windows: check shift operations are available', async function (assert) {
+    assert.expect(9);
+
+    this.data['mail.channel'].records = [
+        { id: 10, last_message: {id: 10} },
+        { id: 20, last_message: {id: 20} },
```

Comment:
Think you can omit `last_message`. In fact, you can just store `channel` and `chat` data in variable and populate init messaging and data the same way:

```js
const channel = {
   channel_type: "channel",
   id: 20,
   name: "General",
};
const chat = {
   channel_type: "chat",
   direct_partner: [{
      id: 7,
      name: "Demo",
   }],
   id: 10,
};
this.data['mail.channel'].records = [channel,dm];
Object.assign(this.data.initMessaging, {
   channel_slots: {
      channel_channel: [channel],
      channel_direct_message: [chat],
   },
});
```

(data of `mail.channel` in mock server is in fact result of channel_info formatter, not actual `mail.channel` fields? Yeah yeah... don't bother with that ^^')

Reply:
Then we need to refactor `message.prettyBody` because as lastMessage is null :

```

Source: | TypeError: Cannot use 'in' operator to search for 'body' in null     at Function.convertData (http://localhost:8069/mail/static/src/messaging/entity/message/message.js:38:24)     at Function.loadPreviews (http://localhost:8069/mail/static/src/messaging/entity/thread/thread.js:292:47)
-- | --
```

Reply:
I think `channel_fetch_preview` receive a list of channel ids and return a list of object containing id of channel and message formatter data of last message. Maybe in practice if channel has no last message (due to not having any messages), it's not in the list. To play safe, I would adapt `static async loadPreviews` so that it doesn't insert message if `.last_message` is falsy.

Reply:
It's due to the fact that the thread preview considers that if there is a `channel_fetch_preview` result then it has a `last_message`. I don't know if in practice this can happen, but if it does then we need to check if the channel has a `last_message` before trying to create it, otherwise we need to create it in the test.

in `thread.js` :
```js
/**
 * Load the previews of the specified threads. Basically, it fetches the
 * last messages, since they are used to display inline content of them.
 *
 * @static
 * @param {mail.messaging.entity.Thread>[]} threads
 */
static async loadPreviews(threads) {
    const channelIds = threads.reduce((list, thread) => {
        if (thread.model === 'mail.channel') {
            return list.concat(thread.id);
        }
        return list;
    }, []);
    const messagePreviews = await this.env.rpc({
        model: 'mail.channel',
        method: 'channel_fetch_preview',
        args: [channelIds],
    }, { shadow: true });
    for (const preview of messagePreviews) {
        const messageData = preview.last_message;
        this.env.entities.Message.insert(
            this.env.entities.Message.convertData(messageData)
        );
    }
}
```

Reply:
Indeed, I would not assume `preview.last_message` exists, so guard here